x-pack/plugins/actions/server/types.ts

```diff
@@ -141,6 +141,18 @@ export interface ActionType<
     connector?: (config: Config, secrets: Secrets) => string | null;
   };
   isSystemActionType?: boolean;
+  /**
+   * Additional Kibana privileges to be checked by the actions framework.
+   * Use it if you want to perform extra authorization checks based on a Kibana feature.
+   * For example, you can define the privileges a users needs to have to execute
+   * a Case or OsQuery system action.
+   *
+   * The list of the privileges follows the Kibana privileges format usually generated with `security.authz.actions.*.get(...)`.
+   *
+   * It only works with system actions and only when executing an action.
+   * For all other scenarios they will be ignored
+   */
+  getKibanaPrivileges?: (args?: { metadata?: Record<string, unknown> }) => string[];
```

Comment:
The reason this is a function that accepts optional `metadata` is because of a use case of the Cases action. The RBAC model of Cases relies on the `owner` of the case. This information cannot be known on Kibana's `setup`. This information will be part of the actions' `params`. On execution, the actions client or the actions executor needs to pass the params to `getKibanaPrivileges` so the cases action type can construct the Kibana privileges based on the owner. For example, `cases:<owner>/create`. I name it `metadata` because it felt that the `params` naming was coupling the `params` with the action type registry (it started having knowledge of it). I don't know what is best. Let me know if you want me to make it more explicit or use a different structure.

Reply:
I actually think it makes more sense to couple it to the expected type of `Params` vs allowing an arbitrary record. Do we envision system actions that derive their privileges from more arbitrary information? Seems safer to type it as `Params` for now and expand to `Record<string,unknown>` if we need to in the future. Since this is all happening in memory, it shouldn't involve any migrations or breaking changes if we change the definition of this parameter in the future right?

Reply:
Thanks, @ymao1 for the feedback. Yes, this is correct. Everything happens in memory. I will couple it to `Params`.